src/buffer/out/textBuffer.cpp

```diff
@@ -551,7 +552,49 @@ bool TextBuffer::IncrementCircularBuffer(const bool inVtMode)
     // to the logical position 0 in the window (cursor coordinates and all other coordinates).
     _renderTarget.TriggerCircling();
 
-    // First, clean out the old "first row" as it will become the "last row" of the buffer after the circle is performed.
+    // First, check the old first row for hyperlink references
+    // If there are any, search the entire buffer for the same reference

```

Comment:
nit: I might move all this to a helper method that's just `_pruneHyperlinks`, just for clarity's sake